client/lib/signup/step-actions.js

```diff
@@ -520,7 +520,7 @@ export function createAccount(
 				if ( oauth2Signup ) {
 					assign( providedDependencies, {
 						oauth2_client_id: queryArgs.oauth2_client_id,
-						oauth2_redirect: queryArgs.oauth2_redirect,
+						oauth2_redirect: response && response.oauth2_redirect.split( '@' )[ 1 ],
```

Comment:
It' probably worth it to guard this statement a bit to prevent uncaught errors.

```suggestion
						oauth2_redirect: get( response, 'oauth2_redirect', '' ).split( '@' )[ 1 ],
```